Newer clang errors when TARGET_OS_* macros are undefined and used
without checking whether they are defined.

--- library.c.orig	2023-05-12 01:21:08
+++ library.c	2025-06-24 02:00:59
@@ -388,7 +388,7 @@ int openconnect_set_reported_os(struct openconnect_inf
 	if (!os) {
 #if defined(__APPLE__)
 #  include <TargetConditionals.h>
-#  if TARGET_OS_IOS
+#  if defined(TARGET_OS_IOS) && TARGET_OS_IOS
 		/* We need to use Apple's boolean "target" defines to distinguish iOS from
 		 * desktop MacOS. See  https://stackoverflow.com/a/5920028 and
 		 * https://github.com/mstg/iOS-full-sdk/blob/master/iPhoneOS9.3.sdk/usr/include/TargetConditionals.h#L64-L71
